docs/source/amp.rst

```diff
@@ -256,6 +256,141 @@ In this case, combine the two layers using :func:`torch.nn.functional.binary_cro
 or :mod:`torch.nn.BCEWithLogitsLoss`.  ``binary_cross_entropy_with_logits`` and ``BCEWithLogits``
 are safe to autocast.
 
+.. _autocast-xpu-op-reference:
+
+XPU Op-Specific Behavior
+------------------------
+The following lists describe the behavior of eligible ops in autocast-enabled regions.
+These ops always go through autocasting whether they are invoked as part of a :class:`torch.nn.Module`,
+as a function, or as a :class:`torch.Tensor` method. If functions are exposed in multiple namespaces,
+they go through autocasting regardless of the namespace.
+
+Ops not listed below do not go through autocasting.  They run in the type
+defined by their inputs.  However, autocasting may still change the type
+in which unlisted ops run if they're downstream from autocasted ops.
+
+If an op is unlisted, we assume it's numerically stable in ``bfloat16``.
+If you believe an unlisted op is numerically unstable in ``bfloat16``,
+please file an issue.
+
+XPU Ops that can autocast to ``bfloat16``
+""""""""""""""""""""""""""""""""""""""""
+
+``__matmul__``,
+``addbmm``,
+``addmm``,
+``addmv``,
+``addr``,
+``baddbmm``,
+``bmm``,
+``chain_matmul``,
+``multi_dot``,
+``conv1d``,
+``conv2d``,
+``conv3d``,
+``conv_transpose1d``,
+``conv_transpose2d``,
+``conv_transpose3d``,
+``GRUCell``,
+``linear``,
+``LSTMCell``,
+``matmul``,
+``mm``,
+``mv``,
+``prelu``,
+``RNNCell``
+
+XPU Ops that can autocast to ``float32``
+""""""""""""""""""""""""""""""""""""""""
+
+``__pow__``,
+``__rdiv__``,
+``__rpow__``,
+``__rtruediv__``,
+``acos``,
+``asin``,
+``binary_cross_entropy_with_logits``,
+``cosh``,
+``cosine_embedding_loss``,
+``cdist``,
+``cosine_similarity``,
+``cross_entropy``,
+``cumprod``,
+``cumsum``,
+``dist``,
+``erfinv``,
+``exp``,
+``expm1``,
+``group_norm``,
+``hinge_embedding_loss``,
+``kl_div``,
+``l1_loss``,
+``layer_norm``,
+``log``,
+``log_softmax``,
+``log10``,
+``log1p``,
+``log2``,
+``margin_ranking_loss``,
+``mse_loss``,
+``multilabel_margin_loss``,
+``multi_margin_loss``,
+``nll_loss``,
+``norm``,
+``normalize``,
+``pdist``,
+``poisson_nll_loss``,
+``pow``,
+``prod``,
+``reciprocal``,
+``rsqrt``,
+``sinh``,
+``smooth_l1_loss``,
+``soft_margin_loss``,
+``softmax``,
+``softmin``,
+``softplus``,
+``sum``,
+``renorm``,
+``tan``,
+``triplet_margin_loss``
+
+XPU Ops that promote to the widest input type
+"""""""""""""""""""""""""""""""""""""""""""""
+These ops don't require a particular dtype for stability, but take multiple inputs
+and require that the inputs' dtypes match.  If all of the inputs are
+``bfloat16``, the op runs in ``bfloat16``.  If any of the inputs is ``float32``,
+autocast casts all inputs to ``float32`` and runs the op in ``float32``.
+
+``addcdiv``,
```

Comment:
This comment also covers tables in prev. sections... Some, actually most, of the operations listed here are currently not supported by XPU backend. As such this document will be confusing for the end user. I suggest that you need to document only those operations which are really supported by XPU. Having placeholders in documentation does not seem as a right thing to do.

| op | Supported? | Link | 
| --- | --- | --- |
| ``addcdiv`` | no | https://github.com/intel/torch-xpu-ops/blob/65c61554aa1023ee99dce101dc743317d705b863/src/aten/XPUFallback.template#L170 |
| ``addcmul`` | no | https://github.com/intel/torch-xpu-ops/blob/65c61554aa1023ee99dce101dc743317d705b863/src/aten/XPUFallback.template#L171 |
| ``atan2`` | no | https://github.com/intel/torch-xpu-ops/blob/65c61554aa1023ee99dce101dc743317d705b863/src/aten/XPUFallback.template#L181 |
| ``bilinear`` | unknown | |
| ``cross`` | unknown | |
| ``dot`` | no | https://github.com/intel/torch-xpu-ops/blob/65c61554aa1023ee99dce101dc743317d705b863/src/aten/XPUFallback.template#L213 |
| ``grid_sample`` | no | https://github.com/pytorch/pytorch/issues/127002 |
| ``index_put`` | yes | https://github.com/intel/torch-xpu-ops/blob/65c61554aa1023ee99dce101dc743317d705b863/yaml/xpu_functions.yaml#L245 |
| ``scatter_add`` | yes | https://github.com/intel/torch-xpu-ops/blob/65c61554aa1023ee99dce101dc743317d705b863/yaml/xpu_functions.yaml#L267 |
| ``tensordot`` | unknown | |